posthog/async_migrations/runner.py

```diff
@@ -163,47 +165,25 @@ def update_migration_progress(migration_instance: AsyncMigration):
         pass
 
 
-def attempt_migration_rollback(migration_instance: AsyncMigration, force: bool = False):
+def attempt_migration_rollback(migration_instance: AsyncMigration):
     """
     Cycle through the operations in reverse order starting from the last completed op and run
     the specified rollback statements.
     """
     migration_instance.refresh_from_db()
-
-    try:
-        ops = get_async_migration_definition(migration_instance.name).operations
-        # if the migration was completed the index is set 1 after, normally we should try rollback for current op
-        current_index = min(migration_instance.current_operation_index, len(ops) - 1)
-        for op_index in range(current_index, -1, -1):
+    ops = get_async_migration_definition(migration_instance.name).operations
+    # if the migration was completed the index is set 1 after, normally we should try rollback for current op
+    current_index = min(migration_instance.current_operation_index, len(ops) - 1)
+    for op_index in range(current_index, -1, -1):
+        try:
             op = ops[op_index]
             execute_op(op, str(UUIDT()), rollback=True)
-    except Exception as e:
-        error = str(e)
-
-        # forced rollbacks are when the migration completed successfully but the user
-        # still requested a rollback, in which case we set the error to whatever happened
-        # while rolling back. under normal circumstances, the error is reserved to
-        # things that happened during the migration itself
-        if force:
-            last_error = f"Force rollback failed with error: {error}"
-            update_async_migration(
-                migration_instance=migration_instance, status=MigrationStatus.Errored, last_error=last_error,
-            )
-
-        if is_email_available():
-            from posthog.tasks.email import send_async_migration_errored_email
+        except Exception as e:
+            error = f"At operation {op_index} rollback failed with error:{str(e)}"
 
-            send_async_migration_errored_email.delay(
-                migration_key=migration_instance.name, time=datetime.now().isoformat(), error=last_error
-            )
-
-            send_alert_to_plugins(
```

Comment:
are we just getting rid of all this?

Reply:
we're getting rid of the concept that we only update last_error, when in force mode (which made force here be a no-op, the API does check we're in completed state). The rest is handled in process_error i.e. sending the email